lib/src/configuration.dart

```diff
@@ -216,24 +47,40 @@ class Configuration {
   /// Realm with a schema that contains objects that differ from their previous 
   /// specification. If the schema was updated and the schemaVersion was not, 
   /// an [RealmException] will be thrown.
-  double? schemaVersion;
+  int get schemaVersion => realmCore.getSchemaVersion(this);
+  set schemaVersion(int value) => realmCore.setSchemaVersion(this, value);
+
+  String get path => realmCore.getConfigPath(this);
+  set path(String value) => realmCore.setConfigPath(this, value);
+}
+
+class SchemaObject {
+  Type type;
+  String get name => type.toString();
```

Comment:
We don't have to support it initially, but we'd eventually want to enable users to map dart types to different Realm tables. In C# this is achieved using an attribute:

```csharp
[MapTo("bar")]
class Foo : RealmObject { }
```

Since we have the `@RealmModel` annotation in dart, we can allow users to use it to override the name of the model - e.g.

```dart
@RealmModel("bar")
class _Foo { }
```

which will generate something like:

```dart
class Foo extends _Foo with RealmObject {
   @override
   static SchemaObject get schema => SchemaObject(Foo, name: "bar")..properties = [ ];
 }
```

Where `SchemaObject` is defined like:

```dart
class SchemaObject { 
  Type type;
  String name;

  SchemaObject(Type type, { String? name = null }) {
    this.type = type;
    this.name = name ?? type.toString();
  }
}
```

Reply:
Yes we could do that. We have already discussed the support for MapTo attribute. This could be applied to properties and whole classes.

Reply:
Appreciate the code samples, but I don't think we need to change anything now. Keep in mind one thing in Dart, unlike .NET, changing a field to a getter is not a breaking change :)

Reply:
resolving this one